Performance Improvements

ikatza opened the pull request:
Changes to reduce the CPU time necessary to simulate the geant4 stage. I believe the individual  commits in this PR are self explanatory, and the changes transparent. 

There's also some styling and minor refactoring.

I've run a few tests on `sbndcode`, these changes half the necessary CPU time. No serious issue has been risen, but this PR definitively needs a full suit of tests.

ikatza committed:
Do binary_search and use unordered_map to improve performance

Since the vectors are sorted is much faster to perform a binary search.
There's no point on using ordered maps.

ikatza committed:
Use non-static version of RNG

This is recommended in here:
https://mu2e.fnal.gov/public/hep/computing_retired/Random.shtml#fireShoot

ikatza committed:
Use multiplication instead of pow(x, 2) in function which runs often

ikatza committed:
Use vectors instead of maps, a major performance improvement

ikatza committed:
Generate VUV params in the initialisation all at once

ikatza committed:
Define some variables as const, whitespace removal, indentation and some renaming

FNALbuild commented:
A new Pull Request was created by  @ikatza (Iker de Icaza) for develop.

It involves the following packages:

larsim


@LArSoft/level-1-managers, @LArSoft/level-2-managers can you please review it and eventually sign? Thanks.

cms-bot commands are listed <a href="https://cdcvs.fnal.gov/redmine/projects/larsoft/wiki/Pull_request_comments_that_trigger_CI_actions">here</a>

FNALbuild commented:
The code-checks are being triggered in jenkins.

FNALbuild commented:
+code-checks  
 for details see 
 https://buildmaster.fnal.gov/buildmaster/job/GithubIntegration/job/run-pr-code-checks//1165/console

ikatza committed:
Missed resizing on committed version

FNALbuild commented:
Pull request #90 was updated. @LArSoft/level-1-managers, @LArSoft/level-2-managers can you please check and sign again.

FNALbuild commented:
The code-checks are being triggered in jenkins.

FNALbuild commented:
+code-checks  
 for details see 
 https://buildmaster.fnal.gov/buildmaster/job/GithubIntegration/job/run-pr-code-checks//1166/console

lgarren commented:
trigger build

The code still needs to be reviewed, but running the CI anyway to provide feedback.

FNALbuild commented:
The tests are being triggered in jenkins.

FNALbuild commented:
+LArSoft tests OK on slf7 for c7:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15230&builds= 
 +build

FNALbuild commented:
+LArSoft tests OK on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15231&builds= 
 +build

FNALbuild commented:
-lariat failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:lariat/view_builds/index?offset=0&builds=lariat_ci/6566&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15231&builds=

FNALbuild commented:
-argoneut tests failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:argoneut/view_builds/index?offset=0&builds=argoneut_ci/6837&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15231&builds=

FNALbuild commented:
-dune tests failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:dune/view_builds/index?offset=0&builds=dune_ci/13137&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15231&builds=

FNALbuild commented:
-uboone tests failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:uboone/view_builds/index?offset=0&builds=uboone_ci/7357&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15231&builds=

FNALbuild commented:
-sbnd tests failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:sbnd/view_builds/index?offset=0&builds=sbnd_ci/9170&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15231&builds=

FNALbuild commented:
-icarus tests failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:icarus/view_builds/index?offset=0&builds=icarus_ci/4226&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15231&builds=

FNALbuild commented:
This pull request is fully signed and it will be merged to develop and built in the next LArSoft release after it passes the integration tests. This pull request will now be reviewed by the release team before it's merged.)

ikatza commented:
They seem to be failing because of some version mismatch:
```
145: ERROR: Version conflict -- dependency tree requires versions conflicting with current setup of product larsoft: version v09_48_00 vs v09_48_01
```


The SBND and ICARUS builds fail because they need https://github.com/SBNSoftware/sbncode/pull/261

The tests from that PR have plenty of warnings, for the most part I don't understand why these warnings are showing.

lgarren commented:
@ikatza apologies, the experiments have not yet caught up to the latest larsoft release.  Will try again next week.

knoepfel commented on larsim/PhotonPropagation/SemiAnalyticalModel.cxx:
```diff
@@ -99,37 +90,37 @@ void SemiAnalyticalModel::Initialization()
     y_v.push_back(elem.second);
   }
-  fL_abs_vuv = std::round(interpolate(x_v, y_v, 9.7, false)); // 9.7 eV: peak of VUV emission spectrum   // TO DO UNHARDCODE FOR XENON 
+  fvuv_absorption_lenght = std::round(interpolate(x_v, y_v, 9.7, false)); // 9.7 eV: peak of VUV emission spectrum   // TO DO UNHARDCODE FOR XENON
```
Suggest re-spelling `lenght` as `length` here and below.

ikatza replied:
Ups! fixed now

knoepfel commented on larsim/PhotonPropagation/SemiAnalyticalModel.cxx:
```diff
@@ -239,12 +231,12 @@ SemiAnalyticalModel::VUVVisibility(geo::Point_t const& ScintPoint, OpticalDetect
   // calculate visibility by geometric acceptance
   // accounting for solid angle and LAr absorbtion length
-  double visibility_geo = std::exp(-1. * distance / fL_abs_vuv) * (solid_angle / (4 * CLHEP::pi));
+  double visibility_geo = std::exp(-1. * distance / fvuv_absorption_lenght) * (solid_angle / (4 * CLHEP::pi));
```
We understand that you did not change the logic here (just the name of the variable).  But is there any possibility of dividing by zero?  Or does the interpolation done above ensure that that can't happen?

ikatza replied:
There is some possibility, depending on the input parameters. I've encapsulated the calculation on a function, check that is not 0 or less and made the variable a `const`.

knoepfel reviewed (commented):
Changes look reasonable, @ikatza.  Only a couple very minor comments/questions.

lgarren commented:
trigger build

FNALbuild commented:
The tests are being triggered in jenkins.

FNALbuild commented:
+LArSoft tests OK on slf7 for c7:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15278&builds= 
 +build

FNALbuild commented:
+LArSoft tests OK on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15277&builds= 
 +build

FNALbuild commented:
-uboone tests failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:uboone/view_builds/index?offset=0&builds=uboone_ci/7373&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15277&builds=

FNALbuild commented:
-lariat failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:lariat/view_builds/index?offset=0&builds=lariat_ci/6582&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15277&builds=

FNALbuild commented:
+argoneut tests OK on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:argoneut/view_builds/index?offset=0&builds=argoneut_ci/6854&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15277&builds=

FNALbuild commented:
-icarus tests failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:icarus/view_builds/index?offset=0&builds=icarus_ci/4249&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15277&builds=

FNALbuild commented:
-sbnd tests failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:sbnd/view_builds/index?offset=0&builds=sbnd_ci/9205&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15277&builds=

FNALbuild commented:
-dune tests warning on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:dune/view_builds/index?offset=0&builds=dune_ci/13162&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15277&builds=

ikatza committed:
Drop unnecesary vars, make some others const, check vuv_absoption is not 0

FNALbuild commented:
Pull request #90 was updated. @LArSoft/level-1-managers, @LArSoft/level-2-managers can you please check and sign again.

FNALbuild commented:
The code-checks are being triggered in jenkins.

FNALbuild commented:
+code-checks  
 for details see 
 https://buildmaster.fnal.gov/buildmaster/job/GithubIntegration/job/run-pr-code-checks//1167/console

lgarren commented:
@ikatza See the icarus and sbnd build failures.  Looks like a matching PR will be needed for them.

ikatza commented:
Hi @lgarren, there is one for both: https://github.com/SBNSoftware/sbncode/pull/261

In the thread there compilation tests are satisfied, and I believe the warnings are mostly unrelated. 

I'm looking at the output of the DUNE tests and it seems there are plenty of warnings for ProtoDUNE, from small variations of products sizes. Not sure if these are meaningful or related to this PR.

DUNE FD seems to be fine.

pgreen135 reviewed (commented):
@ikatza is this attempting to pre-generate all possible transport time parameterisations? This was intentionally avoided as in many cases only a subset are ever required. The overhead from checking the number of dimension is negligible, so I would expect this to be _much_ slower in many circumstances

pgreen135 commented:
@ikatza can you also run the SBND validation suite, and post the results here (or in the SBN PR)

pgreen135 commented:
@lgarren could you not merge this until these checks are complete, as this could significantly impact the simulation in both SBND and DUNE if there is a problem...

ikatza commented:
Hi @pgreen135 

Yes, it pre-generates all parameters. Overhead is compensated if the job has more than a handful of events and/or they're fairly busy events. I suspect that only jobs that generate single MIPs with less than 20 events would be slower. As a reference, it did pass the SBND CI quick single test and that only has two events.

I agree that I shouldn't be merged hastily, and that a full suite of tests are in order.

lgarren commented:
@pgreen135 agreed

@ikatza will run CI tests again with SBNSoftware/sbncode#261

lgarren commented:
trigger build with pull requests #90, SBNSoftware/sbncode#261

FNALbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) SBNSoftware/sbncode#261

FNALbuild commented:
+LArSoft tests OK on slf7 for c7:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15280&builds= 
 +build

FNALbuild commented:
+LArSoft tests OK on slf7 for c7:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15280&builds= 
 +build

FNALbuild commented:
+LArSoft tests OK on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15279&builds= 
 +build

FNALbuild commented:
+LArSoft tests OK on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15279&builds= 
 +build

FNALbuild commented:
-lariat failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:lariat/view_builds/index?offset=0&builds=lariat_ci/6583&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15279&builds=

FNALbuild commented:
-lariat failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:lariat/view_builds/index?offset=0&builds=lariat_ci/6583&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15279&builds=

FNALbuild commented:
-uboone tests failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:uboone/view_builds/index?offset=0&builds=uboone_ci/7374&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15279&builds=

FNALbuild commented:
-uboone tests failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:uboone/view_builds/index?offset=0&builds=uboone_ci/7374&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15279&builds=

FNALbuild commented:
+argoneut tests OK on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:argoneut/view_builds/index?offset=0&builds=argoneut_ci/6855&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15279&builds=

FNALbuild commented:
+argoneut tests OK on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:argoneut/view_builds/index?offset=0&builds=argoneut_ci/6855&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15279&builds=

FNALbuild commented:
-icarus tests warning on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:icarus/view_builds/index?offset=0&builds=icarus_ci/4250&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15279&builds=

FNALbuild commented:
-icarus tests warning on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:icarus/view_builds/index?offset=0&builds=icarus_ci/4250&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15279&builds=

FNALbuild commented:
-dune tests warning on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:dune/view_builds/index?offset=0&builds=dune_ci/13163&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15279&builds=

FNALbuild commented:
-dune tests warning on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:dune/view_builds/index?offset=0&builds=dune_ci/13163&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15279&builds=

FNALbuild commented:
-sbnd tests warning on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:sbnd/view_builds/index?offset=0&builds=sbnd_ci/9206&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15279&builds=

FNALbuild commented:
-sbnd tests warning on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:sbnd/view_builds/index?offset=0&builds=sbnd_ci/9206&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15279&builds=

ikatza committed:
Correction: binary_search() returns true if value is found (which makes sense, duh)

FNALbuild commented:
Pull request #90 was updated. @LArSoft/level-1-managers, @LArSoft/level-2-managers can you please check and sign again.

FNALbuild commented:
The code-checks are being triggered in jenkins.

FNALbuild commented:
+code-checks  
 for details see 
 https://buildmaster.fnal.gov/buildmaster/job/GithubIntegration/job/run-pr-code-checks//1168/console

ikatza commented:
I mistakenly swapped the logic of the `binary_search` algorithm in `SimDriftElectrons_module`. That was quite an omission from my part.

Annoyingly SBND test wasn't sensible to my mistake, fortunately ICARUS was.

On the flip side, it doesn't seem there's any issue related to the light component.

pgreen135 commented:
@ikatza I've run some tests with/without the timing pre-generation commit, and this does seem to be faster this way in most circumstances. The other changes in the light simulation seem sensible, so I am happy with this from my side. I would still suggest running the SBND validation suite, now that we have it -- see https://sbnsoftware.github.io/sbn/sbnci_wiki/sbnci_main for instructions

lgarren commented:
@pgreen135 I'm about to trigger the regular CI here.  Are you able to run the validation with this PR and SBNSoftware/sbncode#261?  Running validation suites is generally the responsibility of the experiment.

lgarren commented:
trigger build with pull requests #90, SBNSoftware/sbncode#261

FNALbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) SBNSoftware/sbncode#261

pgreen135 commented:
@lgarren yes Iker is currently running the validation (had some technical difficulties that Vito is helping with, hence the delay). We will report the results here once its done

lgarren commented:
Thanks @pgreen135

FNALbuild commented:
+LArSoft tests OK on slf7 for c7:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15288&builds= 
 +build

FNALbuild commented:
+LArSoft tests OK on slf7 for c7:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15288&builds= 
 +build

FNALbuild commented:
+LArSoft tests OK on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15287&builds= 
 +build

FNALbuild commented:
+LArSoft tests OK on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15287&builds= 
 +build

FNALbuild commented:
-uboone tests failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:uboone/view_builds/index?offset=0&builds=uboone_ci/7378&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15287&builds=

FNALbuild commented:
-uboone tests failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:uboone/view_builds/index?offset=0&builds=uboone_ci/7378&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15287&builds=

FNALbuild commented:
-lariat failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:lariat/view_builds/index?offset=0&builds=lariat_ci/6587&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15287&builds=

FNALbuild commented:
-lariat failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:lariat/view_builds/index?offset=0&builds=lariat_ci/6587&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15287&builds=

FNALbuild commented:
+argoneut tests OK on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:argoneut/view_builds/index?offset=0&builds=argoneut_ci/6859&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15287&builds=

FNALbuild commented:
+argoneut tests OK on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:argoneut/view_builds/index?offset=0&builds=argoneut_ci/6859&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15287&builds=

FNALbuild commented:
-icarus tests warning on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:icarus/view_builds/index?offset=0&builds=icarus_ci/4256&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15287&builds=

FNALbuild commented:
-icarus tests warning on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:icarus/view_builds/index?offset=0&builds=icarus_ci/4256&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15287&builds=

FNALbuild commented:
-dune tests warning on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:dune/view_builds/index?offset=0&builds=dune_ci/13168&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15287&builds=

FNALbuild commented:
-dune tests warning on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:dune/view_builds/index?offset=0&builds=dune_ci/13168&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15287&builds=

FNALbuild commented:
-sbnd tests warning on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:sbnd/view_builds/index?offset=0&builds=sbnd_ci/9218&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15287&builds=

FNALbuild commented:
-sbnd tests warning on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:sbnd/view_builds/index?offset=0&builds=sbnd_ci/9218&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15287&builds=

lgarren commented:
FYI: I am able to make a local build against the MicroBooNE code.  There are unit test failures that are unrelated to this PR. (MicroBooNE is not yet up to date with  the latest larsoft and wirecell 0.18.0)

ikatza commented:
We're still looking into validation, there has been some more technical difficulties along the way.

lgarren commented:
Thanks for the update @ikatza

ikatza commented:
We have now managed to run the validation tests on SBND. However, I'm just learning that the current CI system is incapable of testing changes that happen at the `g4` stage. People are working to include that capability.

Given the potential impact of these changes, I believe the best is to wait until I'm able to run the full validation test. It's looking like the tests will be ready by early next week.

lgarren commented:
@ikatza No problem.  It's already too late for this week.  ;)

ikatza committed:
Optimise and style PDFastSimPAR and PDFastSimPVS

Quite a lot of changes in this commit.
- Do not use a map for something a vector is enough
- Stop doing so many useless allocations
- Reduce scope of some variables
- Improve initialisation
- Use the message facility
- Stop using double for variables that are int

ikatza committed:
Formatting, function redefinitions, const correctness...

Nothing in here improves performance, just readeability

ikatza committed:
Merge tag 'LARSOFT_SUITE_v09_49_00' into feature/icaza_performance_improvements

$tag_msg

ikatza committed:
Move ifs that skip a loop higher up

knoepfel commented:
@ikatza, is there any update on this PR?

ikatza commented:
The SBN CI system needed to be overhauled to be able to test the g4 stage. The overhauling is in principle finished, but we keep getting errors from various systems. We're still working on it.

I'll update this thread as soon as we have some results.

ikatza committed:
Merge tag 'LARSOFT_SUITE_v09_51_00' into feature/icaza_performance_improvements

Conflicts are mostly due to my mania to clean whitespace

ikatza committed:
Fix compilation issue, and grab missing changes

FNALbuild commented:
Pull request #90 was updated. @LArSoft/level-1-managers, @LArSoft/level-2-managers can you please check and sign again.

FNALbuild commented:
The code-checks are being triggered in jenkins.

FNALbuild commented:
+code-checks  
 for details see 
 https://buildmaster.fnal.gov/buildmaster/job/GithubIntegration/job/run-pr-code-checks//1243/console

ikatza commented:
I finally have some validation tests to show. 

Since the last time I commented in this thread, I pushed some more changes to improve performance in other places. I have tried these changes for DUNE simulations and in some cases these improvements shorten the spent time to a third. 

Importantly, commit [b521f2c](https://github.com/LArSoft/larsim/pull/90/commits/b521f2cef6782cafb8ee905fe91a4fcb3ed4f892) prevents the missing scenarios in which channels with no detected photos were being added to the `OpDetBacktrackerRecord`. I do not know if this omission was the cause for any issues with the PhotonBackTracker, but it does fix the code to match the excepted behaviour. 

Remember that tests need to pick up PR https://github.com/SBNSoftware/sbncode/pull/261, which I have updated to match this version. 

The validation results:

TPC Simulation
https://dbweb9.fnal.gov:8443/LarCI/app/ns:sbnd/build_detail/phase_details?build_id=sbnd_ci/9615&platform=Linux%203.10.0-1160.59.1.el7.x86_64&phase=ci_validation&buildtype=slf7%20e20:prof

PDS Simulation
https://dbweb9.fnal.gov:8443/LarCI/app/ns:sbnd/build_detail/phase_details?build_id=sbnd_ci/9602&platform=Linux%203.10.0-1160.59.1.el7.x86_64&phase=ci_validation&buildtype=slf7%20e20:prof

pgreen135 commented:
Thanks @ikatza for running the validation. From the SBND side I am happy with these changes.

lgarren commented:
trigger build

FNALbuild commented:
The tests are being triggered in jenkins.

FNALbuild commented:
+LArSoft tests OK on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15748&builds= 
 +build

FNALbuild commented:
-lariat failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:lariat/view_builds/index?offset=0&builds=lariat_ci/6743&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15748&builds=

FNALbuild commented:
+argoneut tests OK on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:argoneut/view_builds/index?offset=0&builds=argoneut_ci/7021&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15748&builds=

FNALbuild commented:
+LArSoft tests OK on slf7 for c7:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15747&builds= 
 +build

FNALbuild commented:
-uboone tests failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:uboone/view_builds/index?offset=0&builds=uboone_ci/7543&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15748&builds=

FNALbuild commented:
-icarus tests failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:icarus/view_builds/index?offset=0&builds=icarus_ci/4494&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15748&builds=

FNALbuild commented:
-sbnd tests failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:sbnd/view_builds/index?offset=0&builds=sbnd_ci/9620&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15748&builds=

ikatza commented:
SBND and ICARUS failed because the build didn't include https://github.com/SBNSoftware/sbncode/pull/261

My bad sorry, I didn't link it above. I have edited the comment now.

lgarren commented:
trigger build with pull requests #90, SBNSoftware/sbncode#261

FNALbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) SBNSoftware/sbncode#261

FNALbuild commented:
-dune tests failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:dune/view_builds/index?offset=0&builds=dune_ci/13438&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15748&builds=

FNALbuild commented:
+LArSoft tests OK on slf7 for c7:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15749&builds= 
 +build

FNALbuild commented:
+LArSoft tests OK on slf7 for c7:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15749&builds= 
 +build

FNALbuild commented:
+LArSoft tests OK on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15750&builds= 
 +build

FNALbuild commented:
+LArSoft tests OK on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15750&builds= 
 +build

FNALbuild commented:
-lariat failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:lariat/view_builds/index?offset=0&builds=lariat_ci/6744&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15750&builds=

FNALbuild commented:
-lariat failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:lariat/view_builds/index?offset=0&builds=lariat_ci/6744&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15750&builds=

FNALbuild commented:
+argoneut tests OK on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:argoneut/view_builds/index?offset=0&builds=argoneut_ci/7022&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15750&builds=

FNALbuild commented:
+argoneut tests OK on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:argoneut/view_builds/index?offset=0&builds=argoneut_ci/7022&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15750&builds=

FNALbuild commented:
-uboone tests failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:uboone/view_builds/index?offset=0&builds=uboone_ci/7544&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15750&builds=

FNALbuild commented:
-uboone tests failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:uboone/view_builds/index?offset=0&builds=uboone_ci/7544&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15750&builds=

FNALbuild commented:
-dune tests failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:dune/view_builds/index?offset=0&builds=dune_ci/13439&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15750&builds=

FNALbuild commented:
-dune tests failed on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:dune/view_builds/index?offset=0&builds=dune_ci/13439&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15750&builds=

FNALbuild commented:
-sbnd tests warning on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:sbnd/view_builds/index?offset=0&builds=sbnd_ci/9621&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15750&builds=

FNALbuild commented:
-sbnd tests warning on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:sbnd/view_builds/index?offset=0&builds=sbnd_ci/9621&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15750&builds=

FNALbuild commented:
-icarus tests warning on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:icarus/view_builds/index?offset=0&builds=icarus_ci/4495&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15750&builds=

FNALbuild commented:
-icarus tests warning on slf7 for e20:prof 
 for details see 
 https://lar-ci-history.fnal.gov/LarCI/app/ns:icarus/view_builds/index?offset=0&builds=icarus_ci/4495&builds= 
 for details of the parent CI build see 
 https://lar-ci-history.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/15750&builds=

knoepfel commented:
approve

FNALbuild commented:
This pull request is fully signed and it will be merged to develop and built in the next LArSoft release after it passes the integration tests.
